Userland/Libraries/LibJS/Runtime/Date.cpp

```diff
@@ -285,21 +291,158 @@ double local_tza(double time, [[maybe_unused]] bool is_utc, Optional<StringView>
     return maybe_offset.has_value() ? static_cast<double>(maybe_offset->seconds) * 1000 : 0;
 }
 
-// 21.4.1.8 LocalTime ( t ), https://tc39.es/ecma262/#sec-localtime
+// 21.4.1.7 GetUTCEpochNanoseconds ( year, month, day, hour, minute, second, millisecond, microsecond, nanosecond ), https://tc39.es/ecma262/#sec-getutcepochnanoseconds
+Crypto::SignedBigInteger get_utc_epoch_nanoseconds(i32 year, u8 month, u8 day, u8 hour, u8 minute, u8 second, u16 millisecond, u16 microsecond, u16 nanosecond)
+{
+    // 1. Let date be MakeDay(𝔽(year), 𝔽(month - 1), 𝔽(day)).
+    auto date = make_day(year, month - 1, day);
+
+    // 2. Let time be MakeTime(𝔽(hour), 𝔽(minute), 𝔽(second), 𝔽(millisecond)).
+    auto time = make_time(hour, minute, second, millisecond);
+
+    // 3. Let ms be MakeDate(date, time).
+    auto ms = make_date(date, time);
+
+    // 4. Assert: ms is an integral Number.
+    VERIFY(ms == trunc(ms));
+
+    // 5. Return ℤ(ℝ(ms) × 10^6 + microsecond × 10^3 + nanosecond).
+    auto result = Crypto::SignedBigInteger { ms }.multiplied_by(s_one_million_bigint);
+    result = result.plus(Crypto::SignedBigInteger { static_cast<i32>(microsecond) }.multiplied_by(s_one_thousand_bigint));
+    result = result.plus(Crypto::SignedBigInteger { static_cast<i32>(nanosecond) });
+    return result;
+}
+
+static i64 clip_bigint_to_sane_time(Crypto::SignedBigInteger const& value)
+{
+    static Crypto::SignedBigInteger const min_bigint { NumericLimits<i64>::min() };
+    static Crypto::SignedBigInteger const max_bigint { NumericLimits<i64>::max() };
+
+    // The provided epoch (nano)seconds value is potentially out of range for AK::Time and subsequently
+    // get_time_zone_offset(). We can safely assume that the TZDB has no useful information that far
+    // into the past and future anyway, so clamp it to the i64 range.
+    if (value < min_bigint)
+        return NumericLimits<i64>::min();
+    if (value > max_bigint)
+        return NumericLimits<i64>::max();
+    return value.to_base(10).to_int<i64>().value();
```

Comment:
You could also just get the sign and use `SignedBigInteger::to_u64` so we don't have to go via a String.
(Could also make that a method on SignedBigInteger)

Reply:
Or alternatively just leave a FIXME

Reply:
Added a FIXME - I had the same thought, but this was copy-pasted from an existing Temporal AO, and I didn't wanna deal with unsigned-to-signed overflow here (which is UB, IIRC) :upside_down_face:. It seems odd that `SignedBigInteger` even has a `to_u64` method, so I think a follow-up to make that `to_i64` and use it here is in order.